src/components/Pagination/index.js

```diff
@@ -0,0 +1,360 @@
+import styled from '@emotion/styled'
+import PropTypes from 'prop-types'
+import React, {
+  createContext,
+  forwardRef,
+  useCallback,
+  useContext,
+  useEffect,
+  useImperativeHandle,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+import ActivityIndicator from '../ActivityIndicator'
+import Box from '../Box'
+import Button from '../Button'
+import getPageNumbers from './getPageNumbers'
+import usePaginatedData from './usePaginatedData'
+
+const PaginationContext = createContext()
+export const usePagination = () => useContext(PaginationContext)
+
+// START - Default Components
+const DefaultLeftComponent = () => null
+
+const StyledPageButton = styled(Button, {
+  shouldForwardProp: prop => !['current'].includes(prop),
+})`
+  ${({ current }) =>
+    current
+      ? `
+    text-decoration: underline;
+    &:hover {
+      text-decoration: underline;
+    }
+  `
+      : ''}
+`
+const MAX_PAGE_TAB = 5
```

Comment:
It can be interesting to make it as property of `Pagination` so users can change it if the want

Reply:
If the user want more page he can pass MiddleComponent prop to change the existing one

Reply:
What if he wants same as default but with just more tabs ?